Pod/Classes/LSBinaryCodec.m

```diff
@@ -0,0 +1,324 @@
+//
+//  LSBinaryCodec.m
+//
+
+#import "LSBinaryCodec.h"
+#import "LSSpanContext.h"
+#import "OTTracer.h"
+
+
+@implementation LSBinaryCodec
+
++ (NSData *)encodedMessageForTraceID:(UInt64)traceID
+                              spanID:(UInt64)spanID
+                             baggage:(NSDictionary *)baggage {
+    // Encode our inner message first
+    NSData *inner = [LSBinaryCodec encodeInnerMessageForTraceID:traceID spanID:spanID baggage:baggage];
+
+    // Next, write our outer message by checking the length of the inner message.
+    // TODO: We probably know this capacity, so we should set it:
+    NSMutableData *outer = [[NSMutableData alloc] init];
+
+    // BasicCarrier basic_ctx = 2;
+    [LSBinaryCodec writeTagNumber:2 format:WireFormatMessage buffer:outer];
+    [LSBinaryCodec writeVarintEncodedUInt64:inner.length buffer:outer];
+    [outer appendData:inner];
+
+    return outer;
+}
+
++ (BOOL)decodeMessage:(NSData *)protoEnc
```

Comment:
This also gives us a road to server-side swift integration, which is coming: http://perfect.org/